src/utils/email.js

```diff
@@ -75,58 +77,139 @@ function sendEmail (recipient, subject, html) {
+    utm_content: 'account-verification-email'
+  }
+
+  return appendUtmParams(url, verificationUtmParams)
 }
 
-function getUnsubscribeUrl (subscriber, emailType) {
-  // TODO: email unsubscribe is broken for most emails
-  let url = new URL(`${SERVER_URL}/user/unsubscribe`)
-  const token = (Object.prototype.hasOwnProperty.call(subscriber, 'verification_token')) ? subscriber.verification_token : subscriber.primary_verification_token
-  const hash = (Object.prototype.hasOwnProperty.call(subscriber, 'sha1')) ? subscriber.sha1 : subscriber.primary_sha1
-  url.searchParams.append('token', encodeURIComponent(token))
-  url.searchParams.append('hash', encodeURIComponent(hash))
-  url = appendUtmParams(url, 'unsubscribe', emailType)
-  return url
+// TODO: Unsubscribing from emails is currently only implemented for
+// the monthly unresolved breach report.
+function getUnsubscribeCtaHref ({ subscriber, isMonthlyEmail }) {
+  const path = `${SERVER_URL}/user/unsubscribe`
+  if (!subscriber) {
+    return path
+  }
+
+  if (isMonthlyEmail && !subscriber.primary_verification_token) {
+    throw new Error('subscriber has no primary verification_token')
+  }
+
+  const url = new URL(path)
+  const token = Object.hasOwn(subscriber, 'verification_token')
+    ? subscriber.verification_token
+    : subscriber.primary_verification_token
+  const hash = Object.hasOwn(subscriber, 'sha1')
+    ? subscriber.sha1
+    : subscriber.primary_sha1
+
+  // Mandatory params for unsubscribing a user
+  const unsubscribeUtmParams = {
+    hash,
+    token,
+    utm_campaign: isMonthlyEmail
+      ? 'monthly-unresolved'
+      : 'unsubscribe',
+    utm_content: 'unsubscribe-cta'
+  }
+
+  return appendUtmParams(url, unsubscribeUtmParams)
 }
 
-function getMonthlyUnsubscribeUrl (subscriber, campaign, content) {
-  // TODO: create new subscriptions section in settings to manage all emails and avoid one-off routes like this
-  if (!subscriber.primary_verification_token) throw new Error('subscriber has no primary verification_token')
-  let url = new URL('user/unsubscribe-monthly/', SERVER_URL)
+/**
+ * Check if params contain all mandatory params.
+ *
+ * @param {object} params Params that we like to have checked
+ * @param {string} mandatoryParams A comma separated list of mandatory params
+ * @returns {boolean} True if all mandatory params are present in params
+ */
+function hasMandatoryParams (params, mandatoryParams) {
+  return mandatoryParams.split(',').every(paramKey => {
+    const paramKeyParsed = paramKey.trim()
+    return (
+      Object.hasOwn(params, paramKeyParsed) &&
+      params[paramKeyParsed] !== ''
+    )
+  })
+}
 
-  url = appendUtmParams(url, campaign, content)
-  url.searchParams.append('token', encodeURIComponent(subscriber.primary_verification_token))
+async function postUnsubscribe (req, res) {
+  const data = req.body
+  const { token, utm_campaign: utmCampaign } = data
 
-  return url
+  try {
+    switch (utmCampaign) {
```

Comment:
Re your question in Slack, I would indeed not use this - some people use [extensions](https://connect.mozilla.org/t5/ideas/add-clearurls-to-firefox-for-android-mobile/idc-p/17262) or [browsers](https://support.brave.com/hc/en-us/articles/9982188779405-What-does-Copy-clean-link-mean-) that strip these.